### PR TITLE
MySQL integrations rewrite

### LINKED ISSUE #2240: Rewrite of the MySQL integration article
/cc @alistairmatthews

### LINKED ISSUE #2241: Rewrite of the Pomelo MySQL integration article
/cc @alistairmatthews

### PR DESCRIPTION
**Pull request by alistairmatthews**
## Summary

MySQL and Pomelo MySQL Entity Framework Core integrations updated to match the equivalent SQL Server integrations.

Fixes #2240
Fixes #2241

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/database/mysql-entity-framework-integration.md](https://github.com/dotnet/docs-aspire/blob/41f39dc528e3bc5334bedf85dfec650bb7fbaf05/docs/database/mysql-entity-framework-integration.md) | [.NET Aspire Pomelo MySQL Entity Framework Core integration](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/mysql-entity-framework-integration?branch=pr-en-us-2284) |
| [docs/database/mysql-integration.md](https://github.com/dotnet/docs-aspire/blob/41f39dc528e3bc5334bedf85dfec650bb7fbaf05/docs/database/mysql-integration.md) | [.NET Aspire MySQL database integration](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/mysql-integration?branch=pr-en-us-2284) |
| [docs/toc.yml](https://github.com/dotnet/docs-aspire/blob/41f39dc528e3bc5334bedf85dfec650bb7fbaf05/docs/toc.yml) | [docs/toc](https://review.learn.microsoft.com/en-us/dotnet/aspire/toc?branch=pr-en-us-2284) |

### CODE REVIEW

**Comment by alistairmatthews** [comment]
> @alistairmatthews please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree company="Civica"

**Review by IEvangelist — APPROVED** [review]
This is awesome, thank you @alistairmatthews - only a few things to adjust. Also, could we collapse this into a MySQL TOC, and when expend show the links to the EF Core and then non-EF Core docs. We've done this with a few others that are related.

File: docs/database/includes/mysql-app-host.md [hunk 1]
```diff
@@ -15,4 +19,137 @@ dotnet add package Aspire.Hosting.MySql
...
+
+### Add a MySQL resource with a data volume
+
+To add a data volume to the SQL Server resource, call the <xref:Aspire.Hosting.MySqlBuilderExtensions.WithDataVolume*> method on the SQL Server resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var mysql = builder.AddMySql("mysql")
+                 .WithDataVolume();
```
[the remark this answers is not in the document]
Comment (alistairmatthews) [reply docs/database/includes/mysql-app-host.md]: Fixed.

File: docs/database/includes/mysql-app-host.md [hunk 2]
```diff
@@ -15,4 +19,137 @@ dotnet add package Aspire.Hosting.MySql
...
+
+### Add a MySQL resource with a data bind mount
+
+To add a data bind mount to the MySQL resource, call the <xref:Aspire.Hosting.MySqlBuilderExtensions.WithDataBindMount*> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var mysql = builder.AddMySql("mysql")
+                 .WithDataBindMount(source: @"C:\MySql\Data");
```
[the remark this answers is not in the document]
Comment (alistairmatthews) [reply docs/database/includes/mysql-app-host.md]: Fixed.

File: docs/database/includes/mysql-app-host.md [hunk 3]
```diff
@@ -15,4 +19,137 @@ dotnet add package Aspire.Hosting.MySql
...
+
+### Add a PhpMyAdmin resource
+
+[**phpMyAdmin**](https://www.phpmyadmin.net/) is a popular web-based administration tool for MySQL. You can use it to browse and modify MySQL objects such as databases, tables, views, and indexes. To use phpMyAdmin within your .NET Aspire solution, call the <xref:Aspire.Hosting.MySqlBuilderExtensions.WithPhpMyAdmin*> method. This method adds a new container resource to the solution that hosts phpMyAdmin and connects it to the MySQL container:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var mysql = builder.AddMySql("mysql")
+                 .WithPhpMyAdmin();
```
[the remark this answers is not in the document]
Comment (alistairmatthews) [reply docs/database/includes/mysql-app-host.md]: Fixed.

**Comment by alistairmatthews** [comment]
> This is awesome, thank you @alistairmatthews - only a few things to adjust. Also, could we collapse this into a MySQL TOC, and when expend show the links to the EF Core and then non-EF Core docs. We've done this with a few others that are related.

Thanks @IEvangelist. I've modified the TOC to look like the PostgreSQL one. You might want to check it to make sure I've got it right.

File: docs/toc.yml [hunk 4]
```diff
@@ -193,8 +193,13 @@ items:
...
       - name: MySQL
-        displayName: mysql,database
-        href: database/mysql-integration.md
+        items:
+          - name: MySQL
+            displayName: mysql,database
+            href: database/mysql-integration.md
+          - name: Pomelo MySQL - EF Core
+            displayName: pomelo mysql,database
+            href: database/mysql-entity-framework-integration.md
```
Comment (IEvangelist) [inline docs/toc.yml:202]:
```suggestion
          - name: Pomelo MySQL - EF Core
            displayName: pomelo mysql,database
            href: database/mysql-entity-framework-integration.md
          - name: MySQL
            displayName: mysql,database
            href: database/mysql-integration.md
```

**Comment by davidfowl** [comment]
It would be good to write a general EntityFramework overview. I wonder if we should have an EF node under integrations and all of the ef integrations there 

cc @DamianEdwards @AndriySvyryd

**Pull request merged**